### PR TITLE
add open_ext.lua

### PR DESCRIPTION
**Pull request by takase1121**
This plugin prevents the user from accidentally opening a binary file and break something.
It allows users to open it with other programs (using `xdg-open`).

### CODE REVIEW

**Comment by vincens2005** [comment]
that's cool

**Comment by franko** [comment]
I like the idea but it seems to give a lot of false positive. I am wondering if truncating the line to a fixed number of bytes is not the cause.

Otherwise I would like to have this in the core lite-xl please and the function validate_utf8 should be implemented in C if possible.

**Comment by takase1121** [comment]
> I like the idea but it seems to give a lot of false positive. I am wondering if truncating the line to a fixed number of bytes is not the cause.

This is just to prevent it from reading _too much_ data. I could increase the number, but I'm unsure how well it'll go.

False positive are to expected. Lite XL only supports UTF-8 and UTF-8 only. Suprisingly a lot of binary formats has ASCII/UTF-8 headers or coincidentally *looks* like UTF-8. This could be improved upon in the future.

**Comment by franko** [comment]
I think if you truncate a utf-8 string to an arbitrary number of bytes it can become invalid because you can truncate the sequence of a single unicode character.

There are a lot of false positive and I don't see why this is a plugin.

I would not like to have this as a plugin but I would like to have this into the core lite-xl and with a good robust implementation, preferably with validate_utf8 implemented in C.

Let's not provide half-baked solutions to real problems.

**Comment by takase1121** [comment]
> I think if you truncate a utf-8 string to an arbitrary number of bytes it can become invalid because you can truncate the sequence of a single unicode character.

Good point. I'll fix that.

> Let's not provide half-baked solutions to real problems.
Only if it was as easy as it says. This is heuristics - in layman terms it's guessing. I took advantage of the fact that Lite (and by extension Lite XL) only supports UTF-8. Sure, we can use more complex solutions like reading the magic, but that's [reinventing the wheel](https://man7.org/linux/man-pages/man3/magic_setparam.3.html). libmagic is just overkill.

**Comment by franko** [comment]
> Only if it was as easy as it says. This is heuristics - in layman terms it's guessing. I took advantage of the fact that Lite (and by extension Lite XL) only supports UTF-8. Sure, we can use more complex solutions like reading the magic, but that's [reinventing the wheel](https://man7.org/linux/man-pages/man3/magic_setparam.3.html). libmagic is just overkill.

I agree, this is heuristics and the validate_utf8 you are using is fine to me. I don't mean we should use an heavy-weight implementation or call some library, just ferret the approach you are currently using. If it is heuristics it doesn't mean we don't care about how well or badly it behaves.

I maintain the remarks I already made for this PR.

**Comment by franko** [comment]
I would have liked to integrate this in lite-xl but since nobody is working on that let's add it as a plugin.
